### PR TITLE
feat(docs): improve `@see` meta docstrings

### PR DESCRIPTION
**Pull request by justinmk**

### CODE REVIEW

File: scripts/gen_eval_files.lua [hunk 1]
```diff
@@ -290,8 +282,18 @@ end
 --- Ensure code blocks have one empty line before the start fence and after the closing fence.
 ---
 --- @param x string
+--- @param special string?
+---                | 'see' Normalize for `@see`
```
Comment (justinmk) [inline scripts/gen_eval_files.lua:286]: This is a nice LuaLS feature, not sure if we're using it elsewhere.

File: scripts/gen_eval_files.lua [hunk 2]
```diff
@@ -199,14 +199,6 @@ local function process_params(params)
   return params
 end
 
---- @class vim.gen_vim_doc_fun
---- @field signature string
---- @field doc string[]
---- @field parameters_doc table<string,string>
---- @field return string[]
---- @field seealso string[]
---- @field annotations string[]
```
Comment (justinmk) [inline scripts/gen_eval_files.lua:208]: @lewis6991 do we need this class ? seems like we are using `vim.EvalFn` instead

File: scripts/gen_eval_files.lua [hunk 3]
```diff
@@ -290,8 +282,18 @@ end
...
+--- @param special string?
+---                | 'see' Normalize for `@see`
 --- @return string
-local function norm_text(x)
+local function norm_text(x, special)
+  if special == 'see' then
+    -- @see works better without backticks and parens (for functions).
+    -- "|foo()|" => "foo"
+    -- XXX: LuaLS renders unknown symbols as strikethrough: https://github.com/LuaLS/lua-language-server/blob/8c32d8f95f28f8bc064533031898abfb7dea58b0/script/core/hover/description.lua#L144
+    x = x:gsub([=[|([^() ]+)%(%)|]=], '%1')
```
Comment (justinmk) [inline scripts/gen_eval_files.lua:293]: blocked by https://github.com/LuaLS/lua-language-server/issues/2889 , unless we do some hacks in https://github.com/neovim/neovim/blob/056009f74146b349c66790fbcba3130e85a6c6da/runtime/lua/vim/lsp/util.lua#L1372

File: runtime/lua/vim/_meta/api.lua [hunk 4]
```diff
@@ -122,7 +122,7 @@ function vim.api.nvim__ns_set(ns_id, opts) end
 --- Instruct Nvim to redraw various components.
 ---
 ---
---- @see `:redraw`
+--- @see `:help :redraw`
```
Comment (justinmk) [inline runtime/lua/vim/_meta/api.lua:125]: is this useful?

File: scripts/gen_eval_files.lua (hunk 2 above)
Comment (lewis6991) [reply scripts/gen_eval_files.lua:208]: If it's the same then remove. Can't fully remember but I think I added this as there was a slight difference with one of the fields. I think the other class has more fields that aren't used here.

File: scripts/gen_eval_files.lua (hunk 3 above)
Comment (lewis6991) [reply scripts/gen_eval_files.lua:293]: Using tilde for strike through is GitHub flavoured markdown (as opposed to commonmark), which incidentally is also the reason tilde strike through doesn't work in matrix/element.

I'm not sure which flavour of markdown LSP allows.

**Pull request merged**